config/routes.rb

```diff
@@ -93,13 +93,16 @@
         end
       end
     end
-    resources :teams, except: [:edit, :update, :destroy] do
+    resources :teamsets, only: [:index, :edit, :update] do
       member do
-        patch :dissolve
-      end
-      collection do
-        patch :dissolve_all
+        patch :clone
         patch :randomize
+        patch :dissolve_all
+      end
+      resources :teams, only: [:show] do
```

Comment:
SPOT helps, until it fails you utterly because there's no second source of validation.  It's the belt-and-suspenders counterargument.  Our database fails SPOT -- teamsets have course ids, and teams have both teamset and course ids.  Submissions have assignment and course ids.  Registrations have course and section ids.  Etc.  All of which have been very useful in debugging.  Yes, granted, 100% test coverage (whatever that means) would help too, but we don't have that and won't feasibly have that in the near future.